kubeflow/core/tensorboard.libsonnet

```diff
@@ -0,0 +1,327 @@
+        ]
+      else
+        [
+          $.parts.tb,
+          $.parts.tfDeployment,
+        ],
+  }.all,
+
+  parts:: {
+    // We define the containers one level beneath parts because combined with jsonnet late binding
+    // this makes it easy for users to override specific bits of the container.
+    tbContainerBase:: {
+      name: $.params.name,
+      image: $.params.modelServerImage,
+      imagePullPolicy: "IfNotPresent",
+      args: [
+        "/usr/bin/tensorflow_model_server",
+        "--port=9000",
+        "--model_name=" + $.params.modelName,
+        "--model_base_path=" + $.params.modelPath,
+      ],
+      ports: [
+        {
+          containerPort: 9000,
+        },
+      ],
+      // TODO(jlewi): We should add readiness and liveness probes. I think the blocker is that
+      // model-server doesn't have something we can use out of the box.
+      resources: {
+        requests: {
+          memory: "1Gi",
+          cpu: "1",
+        },
+        limits: {
+          memory: "4Gi",
+          cpu: "4",
+        },
+      },
+    },  // tbContainer
+
+    tbContainer+: $.parts.tbContainerBase +
+                         if $.params.numGpus > 0 then
+                           {
+                             resources+: {
+                               limits+: {
+                                 "nvidia.com/gpu": $.params.numGpus,
+                               },
+                             },
+                           }
+                         else {},
+
+
+    httpProxyContainer:: {
+      name: $.params.name + "-http-proxy",
+      image: $.params.httpProxyImage,
+      imagePullPolicy: "IfNotPresent",
+      command: [
+        "python",
+        "/usr/src/app/server.py",
+        "--port=8000",
+        "--rpc_port=9000",
+        "--rpc_timeout=10.0",
+      ],
+      env: [],
+      ports: [
+        {
+          containerPort: 8000,
+        },
+      ],
+      resources: {
+        requests: {
+          memory: "1Gi",
+          cpu: "1",
+        },
+        limits: {
+          memory: "4Gi",
+          cpu: "4",
+        },
+      },
+    },  // httpProxyContainer
+
+
+    tfDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        name: $.params.name,
+        namespace: $.params.namespace,
+        labels: $.params.labels,
+      },
+      spec: {
+        template: {
+          metadata: {
+            labels: $.params.labels,
+          },
+          spec: {
+            containers: [
+              $.parts.tbContainer,
+              if $.params.httpProxyImage != 0 then
+                $.parts.httpProxyContainer,
+            ],
+            // See:  https://github.com/kubeflow/kubeflow/tree/master/components/k8s-model-server#set-the-user-optional
+            // The is user and group should be defined in the Docker image.
+            // Per best practices we don't run as the root user.
+            securityContext: {
```

Comment:
Delete this as I don't think this securityContext is used by the TensorFlow image